### ISSUE TITLE
rename/refactore plugin `SumCurrents`

### ISSUE THREAD

**Issue by psychocoderHPC** [ComputationalRadiationPhysics/picongpu#783]
- the plugin `SumCurrents` is wrong named and should be called `SumCurrentChange`
- use reduce for sum current change (see plugin [EnergyFields](https://github.com/ComputationalRadiationPhysics/picongpu/blob/master/src/picongpu/include/plugins/EnergyFields.hpp))

**Comment by ax3l** [comment]
What is a `SumCurrentChange`?

I thought this plugin just adds up `J` which is a `SumCurrentDensity`.

**Comment by psychocoderHPC** [comment]
Might be my suggested name is not good, but is it not the change of `J`

**Comment by ax3l** [comment]
Hm, from checking the source on my cell phone, it looks like a simple globsl reduce of fieldJ:

https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/src/picongpu/include/plugins/SumCurrents.hpp

That means the current name "SumCurrent" (or what I proposed "SumCurrentDensity" to be more precise but dbl check factor for that) is valid ;)

**Comment by psychocoderHPC** [comment]
You are right but our `FieldJ` is not the current, it's the change of the current in one time step. Is it right? Might be I am wrong.

**Comment by ax3l** [comment]
Close.
Iur fieldJ is the "currentDensity" which is the change of the *charge*
over time and divided by the volume it went through :)

That's why I proposed to either call it "SumCurrentDensity" or to
multiply it with the cell Volume (if it isn't already) - which makes it
a "SumCurrent".

**Comment by ax3l** [comment]
Close. Our `FieldJ` (as I documented inline long time ago in it's header file) is the "currentDensity" which is the change of the *charge* over time and divided by the volume it went through :) (See [Ampère's circuital law](https://en.wikipedia.org/wiki/Maxwell%27s_equations#Formulation_in_terms_of_electric_and_magnetic_fields) in the Maxwell's equations.) That's why I proposed to either call it "SumCurrentDensity" or to multiply it with the cell Volume (if it isn't already) - which makes it a "SumCurrent".

**Comment by ax3l** [comment]
I think adding a `MPI_Reduce` is already sufficient, so only `rank 0` creates output.
Of course, we can also use the `libPMacc` reduce to reduce the code further.

**Comment by psychocoderHPC** [comment]
OK that's fine, than there is only the refactor part open. :+1:
